src/rdkafka.h

```diff
@@ -7337,8 +7629,209 @@ RD_EXPORT const rd_kafka_topic_partition_list_t *
 rd_kafka_DeleteRecords_result_offsets(
     const rd_kafka_DeleteRecords_result_t *result);
 
-/*
- * DeleteGroups - delete groups from cluster
+/**@}*/
+
+/**
+ * @name Admin API - DescribeGroups
+ * @{
+ */
+
+/**
+ * @brief DescribeGroups result type.
+ *
+ */
+typedef struct rd_kafka_ConsumerGroupDescription_s
+    rd_kafka_ConsumerGroupDescription_t;
+
+/**
+ * @brief Member description included in ConsumerGroupDescription.
+ *
+ */
+typedef struct rd_kafka_MemberDescription_s rd_kafka_MemberDescription_t;
+
+/**
+ * @brief Member assignment included in MemberDescription.
+ *
+ */
+typedef struct rd_kafka_MemberAssignment_s rd_kafka_MemberAssignment_t;
+
+/**
+ * @brief Describe groups from cluster as specified by the \p groups
+ *        array of size \p groups_cnt elements.
+ *
+ * @param rk Client instance.
+ * @param groups Array of groups to describe.
+ * @param groups_cnt Number of elements in \p groups array.
+ * @param options Optional admin options, or NULL for defaults.
+ * @param rkqu Queue to emit result on.
+ *
+ * @remark The result event type emitted on the supplied queue is of type
+ *         \c RD_KAFKA_EVENT_DESCRIBEGROUPS_RESULT
+ */
+RD_EXPORT
+void rd_kafka_DescribeGroups(rd_kafka_t *rk,
+                             const char **groups,
+                             size_t groups_cnt,
+                             const rd_kafka_AdminOptions_t *options,
+                             rd_kafka_queue_t *rkqu);
+
+/**
+ * @brief Get an array of group results from a DescribeGroups result.
+ *
+ * The returned groups life-time is the same as the \p result object.
+ *
+ * @param result Result to get group results from.
+ * @param cntp is updated to the number of elements in the array.
+ */
+RD_EXPORT
+const rd_kafka_ConsumerGroupDescription_t **
```

Comment:
Is the Java client type named ConsumerGroupDescription or GroupDescription?

Reason I'm asking is that it is a bit misaligned with the API named DescribeGroups() but the returned value is ConsumerGroupDescription, and the protocol requests seem to be group-type agnostic.

Reply:
the Java type is named ConsumerGroupDescription. but the functions are called rd_kafka_ListGroups and rd_kafka_DescribeGroups as there was rd_kafka_DeleteGroups already. Also in Java they are called listConsumerGroups, describeConsumerGroups and deleteConsumerGroups. Not sure how to proceed here.

Reply:
Let's use the Java naming with Consumer in the method and type names.
For DeleteGroups we can add an alias function and deprecate the old one.

Reply:
Or just document on DeleteGroups that the Java counterpart is called `DeleteConsumerGroups` so that people will find it when searching, probably enough.

Reply:
done